polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -638,3 +653,289 @@ pub fn start(
 	.run();
 	(to_queue_tx, from_queue_rx, run)
 }
+
+struct Unscheduled {
+	unscheduled: HashMap<PvfExecPriority, VecDeque<ExecuteJob>>,
+	counter: HashMap<PvfExecPriority, usize>,
+}
+
+impl Unscheduled {
+	// A threshold reaching which we reset counted jobs.
+	// The max expected queue_size in normal conditions, at the beginning of each block,
+	// a validator will submit jobs for at least vrf_module_samples(6) + 1 for backing
+	// the parachain candidate they are assigned to, there is some buffer added to cover
+	// for situations, where more work arrives in the queue.
+	const MAX_COUNT: usize = 12;
```

Comment:
This section is a bit confusing. Maybe `SCHEDULING_WINDOW_SIZE` is a better name.